src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Waiting/VisualStudioOperationWaitIndicator.StaticWaitIndicator.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Microsoft.VisualStudio.ProjectSystem.Waiting;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Threading;
+
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Waiting
+{
+    internal partial class VisualStudioOperationWaitIndicator
+    {
+        internal static class StaticWaitIndicator
+        {
+            public static void WaitForOperation(IVsUIService<IVsThreadedWaitDialogFactory> waitDialogFactory, string title, string message, bool allowCancel, Action<CancellationToken> action)
+            {
+                _ = WaitForOperationWithResult(waitDialogFactory, title, message, allowCancel, action);
+            }
+
+            public static T WaitForOperation<T>(IVsUIService<IVsThreadedWaitDialogFactory> waitDialogFactory, string title, string message, bool allowCancel, Func<CancellationToken, T> action)
+            {
+                if (typeof(T) == typeof(Task))
+                    throw new ArgumentException(nameof(T));
+
+                (_, T result) = WaitForOperationWithResult(waitDialogFactory, title, message, allowCancel, action);
+                return result;
+            }
+
+            public static void WaitForAsyncOperation(IVsUIService<IVsThreadedWaitDialogFactory> waitDialogFactory, JoinableTaskContext joinableTaskContext, string title, string message, bool allowCancel, Func<CancellationToken, Task> asyncFunction)
+            {
+                _ = WaitForAsyncOperationWithResult(waitDialogFactory, joinableTaskContext, title, message, allowCancel, asyncFunction);
+            }
+
+            public static T WaitForAsyncOperation<T>(IVsUIService<IVsThreadedWaitDialogFactory> waitDialogFactory, JoinableTaskContext joinableTaskContext, string title, string message, bool allowCancel, Func<CancellationToken, Task<T>> asyncFunction)
+            {
+                (_, T result) = WaitForAsyncOperationWithResult(waitDialogFactory, joinableTaskContext, title, message, allowCancel, asyncFunction);
+                return result;
+            }
+
+            public static WaitIndicatorResult WaitForAsyncOperationWithResult(IVsUIService<IVsThreadedWaitDialogFactory> waitDialogFactory, JoinableTaskContext joinableTaskContext, string title, string message, bool allowCancel, Func<CancellationToken, Task> asyncFunction)
```

Comment:
This takes a Task and doesn't return a result, but ends with 'WithResult' which is confusing to me.